internal/jujuapi/httpproxy.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 Canonical.
+package jujuapi
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/juju/names/v5"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/rpc"
+)
+
+type HTTPProxier struct {
```

Comment:
Interface and struct named the same, this is confusing... Let's refactor some of this together